std/log/logger.ts

```diff
@@ -37,153 +52,58 @@ export class LogRecord {
   }
 }
 
-export interface LoggerOptions {
-  handlers?: BaseHandler[];
-}
-
-export class Logger {
-  #level: LogLevels;
-  #handlers: BaseHandler[];
-  readonly #loggerName: string;
-
-  constructor(
-    loggerName: string,
-    levelName: LevelName,
-    options: LoggerOptions = {},
-  ) {
-    this.#loggerName = loggerName;
-    this.#level = getLevelByName(levelName);
-    this.#handlers = options.handlers || [];
-  }
-
-  get level(): LogLevels {
-    return this.#level;
-  }
-  set level(level: LogLevels) {
-    this.#level = level;
-  }
-
-  get levelName(): LevelName {
-    return getLevelName(this.#level);
-  }
-  set levelName(levelName: LevelName) {
-    this.#level = getLevelByName(levelName);
-  }
-
-  get loggerName(): string {
-    return this.#loggerName;
-  }
-
-  set handlers(hndls: BaseHandler[]) {
-    this.#handlers = hndls;
-  }
-  get handlers(): BaseHandler[] {
-    return this.#handlers;
-  }
-
-  /** If the level of the logger is greater than the level to log, then nothing
-   * is logged, otherwise a log record is passed to each log handler.  `msg` data
-   * passed in is returned.  If a function is passed in, it is only evaluated
-   * if the msg will be logged and the return value will be the result of the
-   * function, not the function itself, unless the function isn't called, in which
-   * case undefined is returned.  All types are coerced to strings for logging.
-   */
-  private _log<T>(
-    level: number,
-    msg: (T extends GenericFunction ? never : T) | (() => T),
-    ...args: unknown[]
-  ): T | undefined {
-    if (this.level > level) {
-      return msg instanceof Function ? undefined : msg;
+const DEFAULT_LOGGER_NAME = "logger";
+
+export class BaseLogger {
+  name: string;
+  logLevel: LogLevel;
+  handlers: Handler[];
+  constructor(logLevel: LogLevel, {
```

Comment:
Currently, when logging to an empty logger it throws an error.

```js
const logger = new Logger();

logger.warn("hello world");
```

I suggest making the whole constructor argument of `BaseLogger` an object, and use the default log level `INFO`.

```js
const DEFAULT_LOG_LEVEL = logLevels.info;

export class BaseLogger {
    /* ... */
    constructor({
        logLevel = DEFAULT_LOG_LEVEL,
        name = DEFAULT_LOGGER_NAME,
        handlers = [],
     }
    /* ... */
}
```